src/pages/ShopCart.tsx

```diff
@@ -1,95 +1,112 @@
-        if(savedCart) {
-            const items = JSON.parse(savedCart);
-            setCartItems(items);
-            calculateTotal(items);
-            setCartCount(items.length);
-        }
-    }, []);
+  useEffect(() => {
+    const savedCart = localStorage.getItem('shoppingCart');
+    if (savedCart) {
+      const items = JSON.parse(savedCart);
+      setCartItems(items);
+      calculateTotal(items);
+      const newCartCount = items.reduce((acc: number, item: Product) => acc + item.quantity, 0);
+      setCartCount(newCartCount);
+    }
+  }, [setCartCount]);
 
-    const calculateTotal = (items: Product[]) => {
-        const total = items.reduce((acc, item) => acc + item.price * item.quantity, 0);
-        setTotalPrice(total);
-    };
+  const calculateTotal = (items: Product[]) => {
+    const total = items.reduce((acc, item) => acc + item.price * item.quantity, 0);
+    setTotalPrice(total);
+  };
 
-    const removeItem = (productId: number) => {
-        const updateCart = cartItems.map((item) => {
-          if (item.id === productId) {
-            if (item.quantity > 1) {
-              // Decrease quantity
-              return { ...item, quantity: item.quantity - 1 };
-            }
-            // Quantity is 1, item will be removed
-            return null;
+  const increaseQuantity = (productId: number) => {
+    const updatedCart = cartItems.map((item) =>
+      item.id === productId ? { ...item, quantity: item.quantity + 1 } : item
+    );
+    updateCartState(updatedCart);
+  };
+
+  const decreaseQuantity = (productId: number) => {
+    const updatedCart = cartItems
+      .map((item) => {
+        if (item.id === productId) {
+          if (item.quantity > 1) {
+            return { ...item, quantity: item.quantity - 1 };
           }
-          return item;
-        })
-        .filter((item) => item !== null);
+          return null; // Remove item if quantity is less than 1
+        }
+        return item;
+      })
+      .filter((item): item is Product => item !== null);
+    updateCartState(updatedCart);
+  };
 
+  const updateCartState = (updatedCart: Product[]) => {
+    setCartItems(updatedCart);
+    localStorage.setItem('shoppingCart', JSON.stringify(updatedCart));
+    calculateTotal(updatedCart);
+    const newCartCount = updatedCart.reduce((acc, item) => acc + item.quantity, 0);
+    setCartCount(newCartCount);
+  };
 
-        setCartItems(updateCart as Product[]);
-        localStorage.setItem("shoppingCart", JSON.stringify(updateCart));
-        calculateTotal(updateCart as Product[]);
-        const newCartCount = (updateCart as Product[]).reduce(
-          (acc, item) => acc + item.quantity,
-          0
-        );
-        setCartCount(newCartCount);
-      };
-    
+  return (
+    <section 
+    id="cart"
+    className="bg-background-light p-4 shadow rounded-t-lg"
+    >
+      <h1 className="text-4xl font-bold mb-4 text-text-primary text-center">Shopping Cart</h1>
```

Comment:
```suggestion
      <h2 className="text-4xl font-bold mb-4 text-text-primary text-center">Shopping Cart</h2>
```
Lets not forget our h1 is in Logo.tsx inside Header.tsx!